### PR TITLE
Rfc on separating

### PR DESCRIPTION
**Pull request by cshoredaniel**
Hi, a bunch more patches; not sure the best way to split them into separate PR's for (presumably) easier management.  Any preferences?

### CODE REVIEW

File: classes/openwrt-kmods.bbclass [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+ROOTFS_POSTUNINSTALL_COMMAND_append += ' openwrt_flatten_modules_hook; '
```
Comment (kraj) [inline classes/openwrt-kmods.bbclass]: We dont need += here, just = is ok

File: classes/openwrt.bbclass [hunk 2]
```diff
@@ -13,3 +13,13 @@ CFLAGS += "-I${STAGING_INCDIR}/lua5.1"
...
+
+VIRTUAL-RUNTIME_dev_manager ?= "procd"
+VIRTUAL-RUNTIME_login_manager ?= "busybox"
+VIRTUAL-RUNTIME_init_manager ?= "procd"
+VIRTUAL-RUNTIME_kmod_manager ?= "ubox"
+VIRTUAL-RUNTIME_syslog ?= "ubox"
+VIRTUAL-RUNTIME_base-utils ?= "busybox"
+VIRTUAL-RUNTIME_network_manager ?= "netifd"
+
+DISTRO_FEATURES_BACKFILL_CONSIDERED_append ?= "sysvinit systemd"
```
Comment (kraj) [inline classes/openwrt.bbclass]: This should not be ?= but =

File: conf/distro/nodistro.conf [hunk 3]
```diff
@@ -0,0 +1,10 @@
+VIRTUAL-RUNTIME_dev_manager ?= "procd"
+VIRTUAL-RUNTIME_login_manager ?= "busybox"
+VIRTUAL-RUNTIME_init_manager ?= "procd"
+VIRTUAL-RUNTIME_kmod_manager ?= "ubox"
+VIRTUAL-RUNTIME_syslog ?= "ubox"
+VIRTUAL-RUNTIME_base-utils ?= "busybox"
+
+DISTRO_FEATURES_BACKFILL_CONSIDERED_append ?= "sysvinit systemd"
```
Comment (kraj) [inline conf/distro/nodistro.conf]: this should not be ?= but =

**Comment by cshoredaniel** [comment]
I've made the requested changes and am running through musl and glibc clean builds (i.e. no cache) - I'll let you know how that goes.

**Comment by cshoredaniel** [comment]
Oh yeah, in my local.conf I've set 'DISTRO_FEATURES += ' procd' to avoid sysvinit or systemd - is that actually necessary?  (I'd have to remove the check for procd distro feature in procd to make it work without)  If it's to be an alternative init_manager / dev_manager it'd probably best to keep it and add it to the list for OE-core?

**Comment by cshoredaniel** [comment]
Status: core-image-minimal, openwrt-image-base and openwrt-image-full now do a clean compile on musl.  Next is to verify clean compile with glibc (no state; it seems some changes were not picked up when doing cached builds, so there some minor corrections added to this series now).

**Comment by cshoredaniel** [comment]
Oh and runqemu qemux86 boots to prompt for cli (and cli works) with core-image-minimal, netifd is working, at least for core functionality on openwrt-image-base, along with some luci (luci needs work though), and with openwrt-image-full mountd launches (but to work needs kernel-module-auto_fs4 which is currently not a build dependency; is there a way to make the build of that module required if mountd is being built (and auto-enable it, rather than having to do local fragments)?

**Pull request merged**